scripts/p2p-test-tool.py

```diff
@@ -0,0 +1,403 @@
+
+
+
+def check_network_convergence(container):
+    peers_metric = ''
+    peers_metric_expected = args.peers_amount
+    timeout = 200
+    count = 0
+
+    while count < 200:
+        cmd = f'curl -s {container.name}:9095'
+        r = container.exec_run(cmd=cmd).output.decode('utf-8')
+        print(r)
+        print(f"checking {count} of {timeout} seconds")
+        # peers_metric = int(float(re.search(r'^peers (.*)$', r, re.MULTILINE).group().rsplit(' ', 1)[1]))
+        for line in r.splitlines():
+            if line == f"peers {args.peers_amount}.0":
+                print("Network converged.")
+                return 0
+        time.sleep(10)
+        count += 10
+    print("Timeout of {timeout} seconds reached ... exiting network convergence pre tests probe.")
+    return 1 
+
+
+def check_network_convergence_alternate(peers_metric_expected):
+    """Alternate check for network convergence before running tests."""
+    # Currently unused but is left in code example purposes
+    print("Checking for network convergence. This could take a while. Max is 200 seconds.")
+    for container in client.containers.list(all=True, filters={"name":f"scan.{args.network}"}):
+        container.remove(force=True, v=True)
+
+    scan_node = {}
+    scan_node['name'] = f"scan.{args.network}"
+    scan_node['volume'] = client.volumes.create()
+    container = client.containers.run('alpine:latest', \
+        name=scan_node['name'], \
+        detach=True, \
+        network=args.network, \
+        volumes={scan_node['volume'].name: {'bind': '/app', 'mode': 'rw'}}, \
+        hostname=scan_node['name'], tty=True)
+    r = container.exec_run(cmd='apk update')
+    r = container.exec_run(cmd='apk add python3')
+    r = container.exec_run(cmd='pip3 install requests')
+
+    # Generating a file out in code to run on docker container. An alternative way.
+    var =("import tempfile\n"
+          "import os\n"
+          "import re\n"
+          "import requests\n"
+          "import time\n"
+          "import subprocess\n"
+          "peers_metric = ''\n"
+          f"peers_metric_expected = {peers_metric_expected}\n"
+          "timeout = 200\n"
+          "count = 0\n"
+
+          "while peers_metric != peers_metric_expected:\n"
+          '    print(f"checking {count} of {timeout} seconds")\n'
+          f"    metrics = requests.get('http://bootstrap.{args.network}:9095').content.decode('utf-8')\n"
+              # peers_metric = float(re.findall(r'^peers (.*)$', metrics, re.MULTILINE)[0]) # alt get peers
+          "    try:\n"
+          "        peers_metric = int(float(re.search(r'^peers (.*)$', metrics, re.MULTILINE).group().rsplit(' ', 1)[1]))\n"
+          "    except Exception as e:\n"
+          "        pass\n"
+          "    if count >= timeout:\n"
+          '        print("timeout of {timeout} seconds reached ... exiting pre tests probe")\n'
+          "        break\n"
+          "    time.sleep(10)\n"
+          "    count += 10\n")
```

Comment:
Why don't you just import this file from a different file?

Reply:
That would be the easiest. I wanted to keep it in the same file. I know it's not practical. We don't even need this. I'll actually remove it out.